src/dune/exe_rules.ml

```diff
@@ -90,19 +90,33 @@ let executables_rules ~sctx ~dir ~expander ~dir_contents ~scope ~compile_info
      stubs, we link the corresponding object files directly. It would be nice to
      make the code more uniform. *)
   let ext_lib = ctx.lib_config.ext_lib in
+  let link_flags_for_relative_foreign_archives () =
```

Comment:
This binding is better placed inside `let link_args`. Because it's not used anywhere else.

Reply:
Fixed in upcoming commit.